### ISSUE TITLE
IP leak when using TC as device VPN and chaining with Orbot

### ISSUE THREAD

**Issue by Maio-msdos** [TrackerControl/tracker-control-android#359]
**Do you use TrackerControl from Google Play/F-Droid/other?**
Fdroid Version 2022.10.25-fdroid

**Describe the bug**
IP leak when using TC as the VPN and chaining with Orbot using the TC Socks5 proxy and port forwarding configuration 

In the first instance I configured TC to send TCP to the Orbot Socks5 proxy using port 9050 as well as port forwarded port 53 UDP to Orbot DNS port 5400, DNS leak test was fine and some IP tests were fine but I consistently got an IP leak when testing with whatsmyip.com

This led me to think that perhaps it was making a port 443 UDP request to dodge the Orbot Socks5 proxy, so I configured TC to port forward all UDP ports from 0-65535 to an unused port (29000) apart from port 53 which I left forwarding to 5400 and port 443 which I forwarded to 9050 (forwarding UDP 443 to 29000 would for some reason also mess with TCP breaking web browsing)

N.B I don't get IP leaks from whatsmyip.com when using Orbot as the VPN

**To Reproduce**
Minimum steps to reproduce the behavior:
1. Run TC as always on VPN and configure TC socks5 proxy to port 9050 and port forward TC UDP port 53 to port 5400
2. Run Orbot with Sock5 proxy and Tor DNS enabled
3. View TC traffic log
4. Browse to whatsmyip.com, to see IP leak

Can send export of TC config if desired, for port forwarding all ports

**Comment by guttermonk** [comment]
I had the same issue with TC and Orbot. Then I tried Invizible Pro instead of Orbot, and still had the same problem. 

Finally, I tried replacing TC with Netguard and configured it to use DNScrypt and Tor using Invizible Pro, per these instructions: https://invizible.net/en/invizible-and-netguard-firewall/

Since I use the default browser that comes with my custom ROM, I had to go into the Netguard settings and check "manage system apps". Apparently this was the source of my IP leak originally, because Netguard was able to manage my browser app while TC had a warning that I could not (even when I had checked "manage system apps" in the TC advanced settings).

**Comment by kasnder** [comment]
Yes. All apps must be included but even then I couldn't offer any guarantees that there are no bugs whatsoever in the app. There's always something.
